modules/Microsoft.Automation/automationAccounts/deploy.bicep

```diff
@@ -11,24 +11,17 @@ param location string = resourceGroup().location
 ])
 param skuName string = 'Basic'
 
-@description('Optional. User identity used for CMK. If you set encryptionKeySource as Microsoft.Keyvault encryptionUserAssignedIdentity is required.')
-param encryptionUserAssignedIdentity string = ''
+@description('Optional. The resource ID of a key vault to reference a customer managed key for encryption from.')
+param cMKKeyVaultResourceId string = ''
 
-@description('Optional. Encryption Key Source. For security reasons it is recommended to use Microsoft.Keyvault if custom keys are available.')
-@allowed([
-  'Microsoft.Automation'
-  'Microsoft.Keyvault'
-])
-param encryptionKeySource string = 'Microsoft.Automation'
+@description('Optional. The name of the customer managed key to use for encryption.')
+param cMKKeyName string = ''
 
-@description('Optional. The name of key used to encrypt data. This parameter is needed only if you enable Microsoft.Keyvault as encryptionKeySource.')
-param keyName string = ''
+@description('Optional. User assigned identity to use when fetching the customer managed key. If not provided, a system-assigned identity can be used - but must be given access to the referenced key vault first.')
```

Comment:
Double check if description should be updated here as well if no systemAssignedIdentity can actually be used

Reply:
Should work: https://docs.microsoft.com/en-us/azure/automation/automation-secure-asset-encryption